int32 Ax
int32 Bx
int32 Ay
int32 By